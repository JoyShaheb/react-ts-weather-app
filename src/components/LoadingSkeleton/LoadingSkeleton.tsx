import React, { FC } from "react";
import { Grid, Skeleton } from "@mui/material";

interface iLoadingSkeleton {
  cards: number;
}

const LoadingSkeleton: FC<iLoadingSkeleton> = ({ cards }) => {
  const cardHeight = 125;
  const animationType = "wave";
  return (
    <Grid container spacing={1}>
      {Array(cards)
        .fill(0)
        .map((_, index) => (
          <Grid item xs={6} key={index}>
            <Skeleton
              animation={animationType}
              variant="rectangular"
              height={cardHeight}
              sx={{
                borderRadius: 1,
              }}
            />
          </Grid>
        ))}
    </Grid>
  );
};

export default LoadingSkeleton;
