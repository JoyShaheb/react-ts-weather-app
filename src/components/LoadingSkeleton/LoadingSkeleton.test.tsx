import React from "react";
import { render, cleanup } from "@testing-library/react";
import LoadingSkeleton from "./LoadingSkeleton";

describe("LoadingSkeleton", () => {
  afterEach(cleanup);

  it("renders correctly", () => {
    const { asFragment } = render(<LoadingSkeleton cards={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
