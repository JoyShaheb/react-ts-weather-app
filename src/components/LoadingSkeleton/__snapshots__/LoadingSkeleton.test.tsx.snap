// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSkeleton renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1k0nuo-MuiSkeleton-root"
        style="height: 125px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1k0nuo-MuiSkeleton-root"
        style="height: 125px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1k0nuo-MuiSkeleton-root"
        style="height: 125px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
